lib/slax_web/websockets/disable.ex

```diff
@@ -0,0 +1,179 @@
+defmodule SlaxWeb.Disable do
+  @moduledoc """
+  A module that handles Slack websocket payloads for disabling Slax interactions and
+  builds modal views for slack https://api.slack.com/reference/surfaces/views
+  """
+
+  alias Slax.Slack
+  alias Slax.Channels
+
+  def handle_payload(%{
+        "trigger_id" => trigger_id,
+        "type" => "shortcut",
+        "callback_id" => "disable_slax"
+      }) do
+    view = build_disable_view(%{trigger_id: trigger_id})
+
+    Slack.open_modal(%{trigger_id: trigger_id, view: view})
+  end
+
+  def handle_payload(%{
+        "trigger_id" => trigger_id,
+        "type" => "shortcut",
+        "callback_id" => "enable_slax"
+      }) do
+    view = build_enable_view(%{trigger_id: trigger_id})
+
+    Slack.open_modal(%{trigger_id: trigger_id, view: view})
+  end
+
+  def handle_payload(%{
+        "trigger_id" => _trigger_id,
+        "type" => "view_submission",
+        "view" => %{
+          "callback_id" => "disable_view",
+          "state" => %{
+            "values" => values
+          }
+        }
+      }) do
+    with %{
+           "channel_select" => %{
+             "selected_option" => %{"text" => %{"text" => name}, "value" => channel_id}
+           }
+         } <-
+           parse_state_values(values) do
+      Channels.create_or_update_channel(channel_id, %{
+        name: name,
+        disabled: true
+      })
+
+      :ok
+    end
+  end
+
+  def handle_payload(%{
+        "trigger_id" => _trigger_id,
+        "type" => "view_submission",
+        "view" => %{
+          "callback_id" => "enable_view",
+          "state" => %{
+            "values" => values
+          }
+        }
+      }) do
+    with %{
+           "channel_select" => %{
+             "selected_option" => %{"text" => %{"text" => name}, "value" => channel_id}
+           }
+         } <-
+           parse_state_values(values) do
+      Channels.create_or_update_channel(channel_id, %{
+        name: name,
+        disabled: false
+      })
+
+      :ok
+    end
+  end
+
+  defp parse_state_values(values) do
+    values
+    |> Map.values()
+    |> Enum.reduce(%{}, fn value, acc -> Map.merge(acc, value) end)
+  end
+
+  defp build_disable_view(%{trigger_id: trigger_id}) do
+    channels =
+      case Slack.get_channels(%{trigger_id: trigger_id}) do
+        [] ->
+          [%{name: "example", id: "example"}]
+
+        channels ->
+          channels
+      end
+
+    %{
+      type: "modal",
+      callback_id: "disable_view",
+      submit: %{
+        type: "plain_text",
+        text: "Disable"
+      },
+      title: %{
+        type: "plain_text",
+        text: "Disable Slax"
+      },
+      blocks: [
+        %{
+          type: "input",
+          element: %{
+            type: "static_select",
```

Comment:
Why not use a channel_select here too?

Reply:
Actually, never mind. I'm not sure what happened here

Reply:
Finally got the actual channel_select drop down to work but once selecting one, the response doesn't have a name of the channel in it, so it might be better to leave it how it is